### ISSUE TITLE
Cannot Generate QA Script

### ISSUE THREAD

**Issue by jack1729** [My-Random-Thoughts/QA-Checks-v4#2]
Using v4 get an error Custom QA  Script generation failed.

I noticed that the script creates a c:\QA\Results if it doesn't exist so I wonder if it is trying to save new script to new location?

**Comment by jack1729** [comment]
PowerShell : AuthorizationManager check failed.
At line:1 char:1
+ PowerShell -NoProfile -NonInteractive -Command {& 'C:\Data\ServerQA\Q ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : SecurityError: (:) [], PSSecurityException
    + FullyQualifiedErrorId : UnauthorizedAccess

**Comment by My-Random-Thoughts** [comment]
The custom QA script is generated in the same folder as the GUI tool and other files.  The C:\QA\Results is just for the results files (HTML, CSV, XML)

The second error looks like you don't have the correct permissions for the action you are trying to do.

**Comment by Paul-Alpha** [comment]
I am getting the exact same error as Jack above. I am a local admin on my machine, ran powershell as administrator. Set execution policy to unrestricted just as a test. Still get the same thing. The default script runs just fine but after running the config tool I can save my custom settings but not generate a new script to run.

EDIT: Ok I just found a bit of a workaround. I just edited the compiler.ps1 on line 196 to include the settings ini that I created with the configuration tool. Ran the compiler and it gave me the updated script to run. Oddly enough after doing that, I needed to make some more changes so I loaded the config tool again and this time it saved the script successfully. I am not sure what would happen if I change the settings file name again. Figured I would let you know.

**Comment by My-Random-Thoughts** [comment]
What version of PowerShell are you using.?
What happens if you run the compiler script on it's own.? - It should display a menu for you to choose a config file.
What are you calling your config files.? - Maybe it's a character issue.?

You can't run the compiler in the ISE, as it will only compile the default-settings.  This is by design (Line 196 as pointed out above).  The reason for this is because the menu does not display correctly and generates pages of errors.

**Comment by Paul-Alpha** [comment]
Powershell version 5.1

When I run the compiler on its own it doesn't give me a menu to choose. It just runs as the default settings. I just used ISE to edit the file which worked. The settings file is just called Alpha no special characters or anything. It did end up naming the final settings file alpha_alpha build.ini

I got it to function with the workaround just figured the information would help.

Thanks,

**Comment by My-Random-Thoughts** [comment]
I am happy you have it working, but I would like to know the cause of the issue.

If the compiler is not showing a menu, that suggests that the settings file is not saved in the settings folder, or doesn't have an .ini extension.

I have not tested on PowerShell 5.1, but i don't see why it shouldn't work

**Comment by My-Random-Thoughts** [comment]
I am going to close this as "no fault found".  When running the compiler via the ISE and not a console window, it will only compile the default-settings file.  I'll add a warning message to the compiler about this.

If you still have an issue outside of this, please open a new ticket and I'll look into it.

**Comment by jack1729** [comment]
2 things I noticed:
1) because this is downloaded from internet, you need to unblock the powershell files whereever you extract them (right click on .ps1 and select unblock)
2) Either sign the ps1 files or set to set-executionpolicy to unrestricted *** IF YOU TRUST THE CODE**
